### DISCUSSION TITLE
Why can't you access globals from tests but you can access a hashtable?

### DISCUSSION THREAD

**Discussion by efie45** [pester/Pester#2018] [Q&A] [answered]
If I put the following in the discovery space or beforeAll block I am able to access ```$table.myVariable``` in any of the test blocks. 
```powershell
$table = @{
  myVariable = 10
}
```
However, regardless of where I put ```$global:myVariable``` I am not able to access it. 

Why is this? Is there a 'proper' way of setting variables in Pester? I'm trying to pull some metadata from my tests to help automate my documentation and this would be really useful. I'm happy using the first example but am interested in how it works.

**Comment by fflaten** [comment, marked as answer]
I can't reproduce this and it should be the opposite in general. Global variables are always available.

`$table` will only be available in interactive execution (calling test-script directly) OR after having dot-sourced the test-script once. Then the variable exists in the PowerShell session and will be inherited into Pester like the global-variable, incl. when using `Invoke-Pester`.
